Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -100,7 +102,10 @@ public void RegisterExtraEventHandlerForFileChange(Action<string> action)
 
         private void OnJobChanged(string jobName)
         {
-            JobListCache = null;
+            lock (jobsListCacheLockObj)
```

Comment:
Would be cleaner to have a `ClearJobListCache()` helper to avoid having to lock from so many places.

Though I'm not sure I understand the purpose of the `lock` here since the `ListJobs` logic uses a local.